File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/KafkaConsumerAssignmentResponse.java
package no.nav.melosys.eessi.controller.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class KafkaConsumerAssignmentResponse {
    private final String topic;
    private final Integer partition;
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/KafkaConsumerAssignmentResponse.kt
package no.nav.melosys.eessi.controller.dto

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class KafkaConsumerAssignmentResponse(
    val topic: String? = null,
    val partition: Int? = null
) {
    class Builder {
        private var topic: String? = null
        private var partition: Int? = null

        fun topic(topic: String?) = apply { this.topic = topic }
        fun partition(partition: Int?) = apply { this.partition = partition }

        fun build() = KafkaConsumerAssignmentResponse(
            topic = topic,
            partition = partition
        )
    }

    companion object {
        @JvmStatic
        fun builder() = Builder()
    }
}
